x/stakeibc/keeper/msg_server_restore_interchain_account_test.go

```diff
@@ -0,0 +1,66 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	_ "github.com/stretchr/testify/suite"
+
+	stakeibc "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+type RestoreInterchainAccountTestCase struct {
+	validMsg stakeibc.MsgRestoreInterchainAccount
+}
+
+func (s *KeeperTestSuite) SetupRestoreInterchainAccount() RestoreInterchainAccountTestCase {
+	s.CreateTransferChannel(HostChainId)
+
+	hostZone := stakeibc.HostZone{
+		ChainId:      HostChainId,
+		HostDenom:    Atom,
+		Bech32Prefix: GaiaPrefix,
+		ConnectionId: ibctesting.FirstConnectionID,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	defaultMsg := stakeibc.MsgRestoreInterchainAccount{
+		Creator:     "creatoraddress",
+		ChainId:     HostChainId,
+		AccountType: stakeibc.ICAAccountType_DELEGATION,
+	}
+
+	return RestoreInterchainAccountTestCase{
+		validMsg: defaultMsg,
+	}
+}
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_Success() {
+	tc := s.SetupRestoreInterchainAccount()
+	msg := tc.validMsg
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
```

Comment:
To test the success case here, I think we'll need to mock out an inactive ICA, then call `RestoreInterchainAccount`, then check that the channel was made active again

Reply:
done